### ISSUE TITLE
Decorators don't work with class methods

### ISSUE THREAD

**Issue by clintonb** [dbkaplan/dry-rest-permissions#21]
The documentation states that global permissions can be implemented as either `@staticmethod` or `@classmethod` on the model. The decorators (e.g. `@unauthenticated_users`) assume the first argument of the global permissions methods is a `Request`. This is true if the methods are static methods. However, when implemented as class methods, the first argument is the class itself.

Either the decorators should be updated, or the documentation should be modified to note this issue and/or remove the mention of `@classmethod`.

**Comment by dbkaplan** [comment]
You're right, the decorators don't work with classmethods. However, you can still use classmethods for global permissions, you just can't also use the decorators on top of that. That being said, you it would be nice to be able to do both. I will look into a solution.

**Comment by timstaley** [comment]
Hi all, just assessing whether this is a good fit for my uses (looks good) and saw this issue. 
This library might come in handy, as it provides a workaround to the inherent Python 2 classmethod decorator issue:
https://wrapt.readthedocs.io/en/latest/
as discussed here:
https://hynek.me/articles/decorators/

HTH.

**Comment by jjlorenzo** [comment]
Sorry if I don't understand deeply the problem, it is not possible to fix this problem?

Looking at the decorators I see that the `is_object_permission` value is used to determine if the `request` it is on `args[1]` instead of `args[0]`

There any problem if it gets generalized to ?
```python
from rest_framework.request import Request
...
request = args[0] if isinstance(args[0], Request) else args[1]
```
Probably we can even use duck-typing to identify the `Request` instead of `isinstance`
